### PR TITLE
Add machines endpoint

### PR DESCRIPTION
**Pull request by pkrull**

### CODE REVIEW

File: swagger.yaml [hunk 1]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+        available:
+          type: boolean
+        key:
+          type: string
+        version:
+          type: string
+        createdDate:
+          type: string
+          format: date-time
+        createdBy:
```
Comment (chadqueen) [inline swagger.yaml]: would propose to not carry forward default modeling for created/last modified. keeping the dates are fine if you guys really feel they are needed, but i hope we will go to a more centralized audit log implementation so createdBy and lastModBy would not be applicable on the model. The two date fields to me can go away as well, but i would be open to keeping them if you really want them

File: swagger.yaml [hunk 2]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+        totalCoolingTime:
+          type: number
+          format: double
+        heaterTemperature:
+          type: integer
+        available:
+          type: boolean
+        key:
+          type: string
+        version:
```
Comment (chadqueen) [inline swagger.yaml]: versioning machines and other objects is difficult to address in the api file, we should discuss this

File: swagger.yaml [hunk 3]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+    properties:
+      id:
+        type: integer
+      # customerId: - doesn't currently exist but needed for custom machine
+      #   type: integer
+      name:
+        type: string
+      description:
+        type: string
+      imageLarge: assets/images/machines/eos_list.jpg
```
Comment (chadqueen) [inline swagger.yaml]: would assume we would kill the images for now, we currently do not use them

File: swagger.yaml [hunk 4]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+          type: number
+          format: double
+        coolingToScanningTimeIncrementRatio:
+          type: integer
+        totalCoolingTime:
+          type: number
+          format: double
+        heaterTemperature:
+          type: integer
+        available:
```
Comment (chadqueen) [inline swagger.yaml]: would active be a better field name than available?

File: swagger.yaml [hunk 5]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+          type: boolean
+        key:
```
Comment (chadqueen) [inline swagger.yaml]: what does the key represent? can descriptions be added to these properties to describe each one?

File: swagger.yaml [hunk 6]
```diff
@@ -110,4 +179,60 @@ definitions:
...
       message:
-        type: string
\ No newline at end of file
+        type: string
+  Machine:
+    type: object
+    properties:
+      id:
+        type: integer
+      # customerId: - doesn't currently exist but needed for custom machine
```
Comment (chadqueen) [inline swagger.yaml]: would need an organizationId instead of customerId i suppose

File: swagger.yaml [hunk 7]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+        type: string
+      description:
+        type: string
+      imageLarge: assets/images/machines/eos_list.jpg
+        type: string
+      imageSmall: assets/images/machines/eos_list.jpg
+        type: string
+      properties:
+        type: object
+        hatchSpacing:
```
Comment (chadqueen) [inline swagger.yaml]: need sub object identification
fieldName:
  type: object
  properties:

File: swagger.yaml [hunk 8]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+      #   type: integer
+      name:
+        type: string
+      description:
+        type: string
+      imageLarge: assets/images/machines/eos_list.jpg
+        type: string
+      imageSmall: assets/images/machines/eos_list.jpg
+        type: string
+      properties:
```
Comment (chadqueen) [inline swagger.yaml]: maybe scientificProperties like tim had mentioned

File: swagger.yaml [hunk 9]
```diff
@@ -110,4 +179,60 @@ definitions:
...
+          format: double
+        coolingToScanningTimeIncrementRatio:
+          type: integer
+        totalCoolingTime:
+          type: number
+          format: double
+        heaterTemperature:
+          type: integer
+        available:
+          type: boolean
```
Comment (chadqueen) [inline swagger.yaml]: boolean was not coming up in my intellisense, it is a valid type right?

File: swagger.yaml [hunk 10]
```diff
@@ -45,18 +45,132 @@ paths:
...
           description: unexpected error
           schema:
             $ref: '#/definitions/Error'
-definitions:
+  /machines:
+    get:
+      description: Get list of machines
+      operationId: getMachineList
+      responses:
+        200:
```
Comment (tsublette) [inline swagger.yaml]: I think we are quoting all responses "200"

File: swagger.yaml [hunk 11]
```diff
@@ -45,18 +45,132 @@ paths:
...
+          description: Not authorized
           schema:
             $ref: '#/definitions/Error'
         default:
           description: unexpected error
           schema:
             $ref: '#/definitions/Error'
-definitions:
+  /machines:
+    get:
```
Comment (tsublette) [inline swagger.yaml]: Need the correct list of paging parameters for GET - see Chad or my PF

File: swagger.yaml [hunk 12]
```diff
@@ -45,18 +45,132 @@ paths:
...
+          schema:
+            $ref: '#/definitions/Error'
+        default:
+          description: unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    post:
+      description: Add a machine. Not available for external consumers.
+      operationId: postMachine
+      parameter: object
```
Comment (tsublette) [inline swagger.yaml]: You need a schema and $ref here - see my post

File: swagger.yaml [hunk 13]
```diff
@@ -45,18 +45,132 @@ paths:
...
+          description: Machine not found (id invalid)
+        default:
+          description: unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      description: Update a machine. Not available for external consumers.
+      operationId: putMachine
+      parameters:
+      - name: id
```
Comment (tsublette) [inline swagger.yaml]: you can remove the ID parameter. It is in the machine object farther down

**Pull request merged**